bokeh/sphinxext/bokeh_plot.py

```diff
@@ -241,26 +238,31 @@ def process_source(self, source, path, js_filename):
 
         return (script_tag, js_path, source, docstring, height_hint)
 
-    def process_sampledata(self, targetnode, source):
-
-        if not hasattr(self.env, 'all_sampledata_xrefs'):
-            self.env.all_sampledata_xrefs = []
-
-        regex = "(:|bokeh\.)sampledata(:|\.| import )\s*(\w+(\,\s*\w+)*)"
-        matches = re.findall(regex, source)
-        if matches:
-            keywords = set()
-            for m in matches:
-                keywords.update(m[2].replace(" ","").split(','))
-            for keyword in keywords:
-                xref_node = sampledata_node('')
-                self.state.nested_parse(self.content, self.content_offset, xref_node)
-
-                self.env.all_sampledata_xrefs.append({
-                    'docname': self.env.docname,
-                    'target': targetnode,
-                    'keyword': keyword
-                })
+    def process_sampledata(self, source):
+
+        file, lineno =  self.get_source_info()
+        # at the moment only links to the gallery are wanted
```

Comment:
FYI the reason for "only gallery" was because linking into the middle of narrative docs pages was not ideal. If we are generating standalone pages, with source and metadata, for every example, there is no reason the sampledata xfrefs can't link *those*.